pkg/apis/core/validation/validation.go

```diff
@@ -5503,6 +5503,26 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Invalid(specPath, newPod.Status.QOSClass, "Pod QOS Class may not change as a result of resizing"))
 	}
 
+	// TODO(#128675): Do not allow removing resource requests/limits for Burstable Pods.
+	// TODO: Check for sidecar containers too.
+	for ix, ctr := range oldPod.Spec.Containers {
+		resourcesRemoved := func(resourceList, oldResourceList core.ResourceList) bool {
+			if oldResourceList != nil && resourceList == nil {
```

Comment:
```suggestion
			if len(oldResourceList) > len(resourceList) {
```

Reply:
Done.

Reply:
Wait, I think we still need the per-resource checks. Otherwise, you could remove a resource and add a different one in the same request.

Reply:
Added it back.